clients/src/main/java/org/apache/kafka/common/record/FileRecords.java

```diff
@@ -53,6 +50,15 @@
         }
     };
 
+    // mutable state
```

Comment:
The two fields below this are `final`, so they aren't mutable. I assume this comment is just misplaced?